app/controllers/resources_controller.rb

```diff
@@ -38,8 +38,42 @@ def suggestions
     render json: ToolFilterService.new(params).call, include: params[:include], fields: field_params, status: :ok
   end
 
+  def publish_translation
+    if valid_publish_params?
+      render json: publish_translations, status: :ok
+    else
+      render json: {errors: {"errors" => [{source: {pointer: "/data/attributes/id"}, detail: "Record not found."}]}}, status: :unprocessable_entity
+    end
+  end
+
   private
 
+  def valid_publish_params?
+    params.dig("data", "relationships", "languages", "data") && params["resource_id"]
+  end
+
+  def publish_translations
+    translations = []
+    languages = params["data"]["relationships"]["languages"]["data"]
+
+    languages.each do |lang|
+      translations << publish_translation_for_language(lang)
+    end
+    translations
+  end
+
+  def publish_translation_for_language(language_data)
+    translation = find_latest_translation(language_data["id"])
```

Comment:
this should be find or create a draft translation. right now the scope you are using is just finding the most recent translation whether it's been published or not. In reality you should looking for a translation that has `is_published = false`, and if that doesn't exist create a new Translation in that state.

Reply:
ok, understood!